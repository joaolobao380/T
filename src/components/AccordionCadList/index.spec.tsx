import { render, fireEvent, waitFor } from '@testing-library/react-native';
import React from 'react';
import { Text } from 'react-native';

import { AccordionCadList } from '.';

describe('AccordionCadList Component', () => {
  it('Should renders the title correctly', () => {
    const title = 'Test Title';
    const { getByText } = render(
      <AccordionCadList title={title}>
        <Text>Accordion Content</Text>
      </AccordionCadList>
    );

    expect(getByText(title)).toBeTruthy();
  });

  it('toggles accordion on press', async () => {
    const { getByTestId, queryByText } = render(
      <AccordionCadList title="Test Title">
        <Text>Accordion Content</Text>
      </AccordionCadList>
    );

    const toggleButton = getByTestId('accordion_toggle_button');
    fireEvent.press(toggleButton);
    await waitFor(() => {
      expect(queryByText('Accordion Content')).toBeTruthy();
    });

    fireEvent.press(toggleButton);
    await waitFor(() => {
      expect(queryByText('Accordion Content')).toBeNull();
    });
  });

  it('Renders correctly Accordion', () => {
    const { toJSON } = render(
      <AccordionCadList title="Test Title">
        <Text>Accordion Content</Text>
      </AccordionCadList>
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
