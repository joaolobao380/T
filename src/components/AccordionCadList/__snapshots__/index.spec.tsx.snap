// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionCadList Component Renders correctly Accordion 1`] = `
<View
  style={
    {
      "backgroundColor": "#EBEBEB",
      "borderRadius": 8,
      "padding": 16,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="accordion_toggle_button"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View>
        <Text
          style={
            {
              "fontFamily": "Poppins-Regular",
            }
          }
        >
          Test Title
        </Text>
      </View>
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="24"
        bbWidth="24"
        fill="none"
        focusable={false}
        height={24}
        meetOrSlice={0}
        minX={0}
        minY={0}
        stroke="#444444"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={2}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 24,
              "width": 24,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4282664004,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth="2"
        >
          <RNSVGPath
            d="m6 9 6 6 6-6"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "payload": 4282664004,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
</View>
`;
